mongorestore/mongorestore.go

```diff
@@ -396,6 +399,10 @@ func (restore *MongoRestore) Restore() Result {
 			}
 			return Result{Err: fmt.Errorf("mongorestore target '%v' invalid: %v", restore.TargetDirectory, err)}
 		}
+		if err = restore.ReadPreludeMetadata(target); err != nil {
+			// don't error out here because older dump versions will not prelude.json
```

Comment:
Here it’d be better to make ReadPreludeMetadata return (bool, error) to differentiate nonexistence from a bona-fide failure.

Reply:
If EACCES or other “unexpected” cases happen, they need to take down the process. Thus, I think “don’t error out here …” doesn’t work.

If we want to _ignore_ the nonexistence case, then yeah we need to handle that specially.